Source/cryptalgo/SecureSocketPort.cpp

```diff
@@ -212,29 +222,26 @@ void SecureSocketPort::Handler::ValidateHandShake() {
 }
 
 void SecureSocketPort::Handler::Update() {
-    if (IsOpen() == true) {
-        int result;
-
+    if (IsOpen() == true && _ssl != nullptr) {
```

Comment:
in the  SecureSocketPort::Handler::Initialize() an error is returned if _ssl was not constructed. Please code defensively, just ASSERT here on an (_ssl != nullptr);